scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -90,32 +128,42 @@
-                                 [:expected/running :pod/failed] (pod-has-just-completed existing-state-dict)
+                                 [:expected/running :pod/failed] (pod-has-just-completed kcc existing-state-dict)
                                  [:expected/completed :pod/failed] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
                                  [:expected/completed :missing] nil ; Cause it to be deleted.
+                                 [:expected/killed :pod/waiting] (kill-task api-client expected-state-dict pod)
                                  [:expected/killed :pod/running] (kill-task api-client expected-state-dict pod) ; TODO: Where does the datomic update occur? Do we do it when we do [expected/killed :pod/failed], to be similar to mesos, we update it only when the backend says its dead?
 
-                                 [:expected/killed :pod/succeeded] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
+                                 [:expected/killed :pod/succeeded] (do
+                                                                     (pod-has-just-completed kcc existing-state-dict)
+                                                                     (remove-finalization-if-set-and-delete api-client expected-state-dict pod))
                                  [:expected/killed :pod/failed]
-                                 (do ; TODO: Invoke handle-status-update.
+                                 (do
+                                   (pod-has-just-completed kcc existing-state-dict)
                                    (remove-finalization-if-set-and-delete api-client expected-state-dict pod))
-                                 [:expected/killed :missing] nil
+                                 [:expected/killed :missing] (pod-was-killed kcc existing-state-dict)
                                  ; TODO: Implement :pod/unknown cases.
                                  ; TODO: Implement :pod/pending cases.
                                  ; TODO: Implement :pod/need-to-fail cases.
                                  [:missing :pod/running] (kill-task api-client expected-state-dict pod)
                                  [:missing :pod/pending] (kill-task api-client expected-state-dict pod)
                                  [:missing :pod/need-to-fail] (kill-task api-client expected-state-dict pod)
                                  [:missing :pod/succeeded] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
-                                 [:missing :pod/failed] (remove-finalization-if-set-and-delete api-client expected-state-dict pod))]
+                                 [:missing :pod/failed] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
+                                 [:missing :missing] nil
```

Comment:
Put a comment that this state can be seen because of the recur at the end.

Reply:
Done